modules/javafx.controls/src/test/java/test/javafx/scene/control/ControlPropertiesTest.java

```diff
@@ -0,0 +1,220 @@
+import com.sun.javafx.scene.control.InputField;
+import com.sun.javafx.scene.control.IntegerField;
+import com.sun.javafx.scene.control.WebColorField;
+import com.sun.javafx.scene.control.skin.FXVK;
+
+/**
+ * Tests contract for properties and their accessors in the Control type hierarchy.
+ *
+ * Currently uses a list of classes, so any new Controls must be added manually.
+ * Perhaps the test should scan classpath and find all the Controls automagically.
+ */
+public class ControlPropertiesTest {
+
+    private static final boolean FAIL_FAST = !true;
+
+    // list all descendants of Control class.
+    // or perhaps collect all classes in a package as described here:
+    // https://stackoverflow.com/questions/28678026/how-can-i-get-all-class-files-in-a-specific-package-in-java
+    private Set<Class> allControlClasses() {
+        return Set.of(
+            Accordion.class,
+            ButtonBar.class,
+            ButtonBase.class,
+            Button.class,
+            Cell.class,
+            CheckBox.class,
+            CheckBoxListCell.class,
+            CheckBoxTableCell.class,
+            CheckBoxTreeCell.class,
+            CheckBoxTreeTableCell.class,
+            ChoiceBox.class,
+            ChoiceBoxTreeCell.class,
+            ChoiceBoxTreeTableCell.class,
+            ColorPicker.class,
+            ComboBox.class,
+            ComboBoxBase.class,
+            ComboBoxTreeCell.class,
+            ComboBoxTreeTableCell.class,
+            DateCell.class,
+            DatePicker.class,
+            DoubleField.class,
+            FXVK.class,
+            //HTMLEditor.class,
+            Hyperlink.class,
+            IndexedCell.class,
+            InputField.class,
+            IntegerField.class,
+            Labeled.class,
+            Label.class,
+            ListCell.class,
+            ListView.class,
+            MenuBar.class,
+            MenuButton.class,
+            Pagination.class,
+            PasswordField.class,
+            ProgressIndicator.class,
+            ProgressBarTreeTableCell.class,
+            RadioButton.class,
+            ScrollBar.class,
+            ScrollPane.class,
+            Separator.class,
+            SeparatorMenuItem.class,
+            Slider.class,
+            Spinner.class,
+            SplitPane.class,
+            TableRow.class,
+            TableView.class,
+            TabPane.class,
+            TextArea.class,
+            TextField.class,
+            TextFieldTreeCell.class,
+            TextFieldTreeTableCell.class,
+            TitledPane.class,
+            ToggleButton.class,
+            TreeCell.class,
+            TreeTableCell.class,
+            TreeTableRow.class,
+            TreeTableView.class,
+            WebColorField.class
+        );
+    }
+
+    /**
+     * Tests for missing final keyword in Control properties and their accessors.
+     */
+    @Test
+    public void testMissingFinalMethods() {
+        for (Class c: allControlClasses()) {
```

Comment:
I think we use a space before `:`. Don't know if it's being enforced. If you change it there are 2 more places.

Reply:
Adding superfluous spaces increases our carbon footprint ;-)
Modified the formatter to increase the footprint.